frontend/src/components/Profile/TaskDetail.js

```diff
@@ -20,16 +20,40 @@ const TaskDetail = (props) => {
           if (res.ok) {
             return res.json();
           } else {
-            throw new Error(`Cannot fetch task detail of ${taskName}`);
+            throw new Error("Cannot fetch task duration");
           }
         })
-        .then((data) => {
-          setTaskDetail(data[taskName][0]);
-        })
         .catch((err) => {
           console.log(err);
         });
     }
+  };
+
+  const fetchTaskHandler = async () => {
+    try {
+      const token = localStorage.getItem("token");
+      if (token) {
+        const response = await fetch(`${BACKEND_API}/task/${taskName}`, {
+          headers: {
+            Authorization: "Bearer " + token,
+          },
+        });
+        if (!response.ok)
+          throw new Error(`Cannot fetch task detail of ${taskName}`);
+        const data = await response.json();
+        if (data[taskName][0].finish_time) {
```

Comment:
you should check whether
1. `data` has the taskName property.
2. `data[taskName]` is an array with a length of 1 or more